cmd/process-agent/main_common.go

```diff
@@ -100,7 +100,7 @@ func runAgent(globalParams *command.GlobalParams, exit chan struct{}) {
 
 	config.InitRuntimeSettings()
 
-	cfg, err := config.NewAgentConfig(loggerName, globalParams.ConfFilePath, syscfg)
+	_, err = config.NewAgentConfig(loggerName, globalParams.ConfFilePath)
```

Comment:
Is there any way you could extract out the logger code from here? `NewAgentConfig` doesn't seem to actually do anything anymore (other than initializing the logger), and it's not obvious why it's called. Additionally, calling a constructor for it's side effects has always seemed kind of an anti-pattern to me.

Reply:
If it's lots of work though feel free to handle it in a separate PR

Reply:
Yes, these are the side-effects I am going to look into next, just used because of these side-effects. 😄